tests/rules/no-space-in-emphasis.test.js

```diff
@@ -0,0 +1,1025 @@
+		| strike| text ~text~ text          |
+		`,
+		dedent`
+		| Table | Table |
+		| ----- | ----- |
+		| x * y | x * y |
+		| x** y | x** y |
+		| x _ y | x _ y |
+		| x__ y | x__ y |
+		`,
+		dedent`
+		\`\`\`yaml /* autogenerated */
+		# YAML...
+		\`\`\`
+		`,
+		"new_value from *old_value* and *older_value*.",
+		":ballot_box_with_check: _Emoji syntax_",
+		"some_snake_case_function() is _called_",
+		"_~/.ssh/id_rsa_ and _emphasis_",
+		"Partial *em*phasis of a *wo*rd.",
+		dedent`
+		<p>
+		Emphasis inside * HTML * content
+		</p>
+		`,
+		'Emphasis <p data="inside * attribute * content"></p>',
+		"Embedded underscore is okay:\nText _emphas_i_s_ text _emphasis_",
+		"Text *emphasis\nemphasis* text",
+		"Text *emphasis* *emphasis\nemphasis* *emphasis* text",
+		"Text *emphasis* text *emphasis\nemphasis* text *emphasis* text",
+		"Text *emphasis* *emphasis\nemphasis* *emphasis* *emphasis\nemphasis* text *emphasis\nemphasis* text *emphasis* text",
+		"Text text\ntext *emphasis\nemphasis emphasis\nemphasis* text\ntext text",
+		"Text * asterisk",
+		"* Item *emphasis* item\n* Item *emphasis* item\n* Item *emphasis\n  emphasis* item\n* Item *emphasis* item",
+		"* Item * asterisk\n* Item * asterisk",
+		"Emphasis `inside\nof * code *\nblocks` is okay.",
+		"Emphasis `* inside`\ncode\n`blocks *` is okay.",
+		"Emphasis `inside *`\ncode\n`* blocks` is okay.",
+		"Emphasis `inside\n_ code _\nblocks` is okay.",
+		"Emphasis `_ inside`\ncode\n`blocks _` is okay.",
+		"Emphasis `inside _`\ncode\n`_ blocks` is okay.",
+		"Mixed `code_span`\nscenarios\nare _also_ okay.",
+		"Mixed `code*span`\nscenarios\nare *also* okay.",
+		"This paragraph\ncontains *a* mix\nof `*` emphasis\nscenarios and *should*\nnot trigger `*` any\nviolations at *all*.",
+		"This paragraph\ncontains `a * slightly\nmore complicated\nmulti-line emphasis\nscenario * that\nshould * not trigger\nviolations * either`.",
+		"Escaped asterisks \\* should \\* be ignored.",
+		"Escaped asterisks \\* should * be ignored.",
+		"Escaped asterisks * should \\* be ignored.",
+		"Escaped underscores \\_ should \\_ be ignored.",
+		"Escaped underscores \\_ should _ be ignored.",
+		"Escaped underscores _ should \\_ be ignored.",
+		"Escaped asterisks \\** should ** be ignored.",
+		"Escaped asterisks *\\* should ** be ignored.",
+		"Escaped underscores \\__ should __ be ignored.",
+		"Escaped underscores _\\_ should __ be ignored.",
+		"Escaped asterisks ** should \\** be ignored.",
+		"Escaped asterisks ** should *\\* be ignored.",
+		"Escaped underscores __ should \\__ be ignored.",
+		"Escaped underscores __ should _\\_ be ignored.",
+		"Escaped tildes \\~ should \\~ be ignored.",
+		"Escaped tildes \\~~ should ~~ be ignored.",
+	],
+	invalid: [
+		{
+			code: "Broken * emphasis * with spaces",
+			output: "Broken *emphasis* with spaces",
+			errors: [
+				{
+					messageId: "spaceInEmphasis",
+					line: 1,
+					column: 8,
+					endLine: 1,
+					endColumn: 20,
+				},
+			],
+		},
```

Comment:
Markdownlint reports two errors in this case, separate for both prefix & suffix highlighting exactly where the whitespace is instead of highlighting the whole text. Should we do the same? It's better in case of long paragraphs.

<img width="554" alt="Screenshot 2025-06-23 at 2 41 54 PM" src="https://github.com/user-attachments/assets/12444704-b857-4359-91e5-e0d5b9d32e3b" />